lib/client/session.go

```diff
@@ -551,15 +551,15 @@ func (ns *NodeSession) runCommand(ctx context.Context, mode types.SessionPartici
 	return ns.regularSession(ctx, func(s *ssh.Session) error {
 		var err error
 
-		runContext, cancel := context.WithCancel(ctx)
+		runDone := make(chan error)
 		go func() {
-			defer cancel()
 			err = s.Run(strings.Join(cmd, " "))
+			runDone <- err
```

Comment:
```suggestion
			runDone <- s.Run(strings.Join(cmd, " "))
```

and drop the `err` declaration above?